src/hal/SensorInterface.h

```diff
@@ -1,9 +1,14 @@
 #ifndef SENSOR_H
 #define SENSOR_H
 
+template <typename SensorReturnType, typename SensorInputType>
+
 class SensorInterface {
 public:
-    virtual double GetAngle() = 0;
+    virtual SensorReturnType GetValue(SensorInputType) = 0;
```

Comment:
Please, document this interface. This is a good time to add the extension: cschlosser.doxdocgen (you need to add this ID inside ./.devcontainer/devcontainer.json)
This is really helpful, you just type \\ + Tab and it generates the doc for you, you just need to tweak it.

Reply:
I will create another issue/pr specifically for that, otherwise I think this one will become too big/hard to keep track of.